getting-started/background/index/main.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Command index is an HTTP app that displays all previous translations
+// (stored in Firestore) and has a form to request new translations. On form
+// submission, the request is sent to Pub/Sub to be processed in the background.
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"cloud.google.com/go/firestore"
+	"cloud.google.com/go/pubsub"
+	"github.com/GoogleCloudPlatform/golang-samples/getting-started/background"
+)
+
+// topicName is the Pub/Sub topic to publish requests to. The Cloud Function to
+// process translation requests should be subscribed to this topic.
+const topicName = "translate"
+
+// An app holds the clients and parsed templates that can be reused between
+// requests.
+type app struct {
+	pubsubClient    *pubsub.Client
+	pubsubTopic     *pubsub.Topic
+	firestoreClient *firestore.Client
+	tmpl            *template.Template
+}
+
+// newApp creates a new app.
+func newApp(projectID, templateDir string) (*app, error) {
+	ctx := context.Background()
+
+	pubsubClient, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return nil, fmt.Errorf("pubsub.NewClient: %v", err)
+	}
+
+	pubsubTopic := pubsubClient.Topic(topicName)
+
+	firestoreClient, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		return nil, fmt.Errorf("firestore.NewClient: %v", err)
+	}
+
+	// Template referenced relative to the module/app root.
+	tmpl, err := template.ParseFiles(filepath.Join(templateDir, "index.html"))
+	if err != nil {
+		return nil, fmt.Errorf("template.New: %v", err)
+	}
+
+	return &app{
+		pubsubClient: pubsubClient,
+		pubsubTopic:  pubsubTopic,
+
+		firestoreClient: firestoreClient,
+		tmpl:            tmpl,
+	}, nil
+}
+
+func main() {
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	if projectID == "" {
+		log.Fatalf("GOOGLE_CLOUD_PROJECT must be set")
+	}
+
+	a, err := newApp(projectID, "index")
```

Comment:
If it's just going to be "index" in actual code, do you need the second argument?